.ci/packaging.groovy

```diff
@@ -250,3 +249,18 @@ def runIfNoMainAndNoStaging(Closure body) {
     body()
   }
 }
+
+def smartGitCheckout() {
```

Comment:
Candidate to be added to the shared library?

Reply:
I was tempted initially

If for some reason I need to apply the same pattern  somewhere else then I'll refactor it  for sure